### PR TITLE
Refactor/split specs

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/provokateurin/nextcloud-neon/issues/419

### CODE REVIEW

File: packages/dynamite/dynamite/lib/src/openapi_builder.dart [hunk 1]
```diff
@@ -247,7 +247,29 @@ class OpenAPIBuilder implements Builder {
...
+                        ${prefix}Client(
+                          client.baseURL,
+                          baseHeaders: client.baseHeaders,
+                          httpClient: client.httpClient,
+                          cookieJar: client.cookieJar,
+                          authentications: client.authentications,
+                        )
+                        '''),
+                    ),
+                  ]);
```
Comment (Leptopoda) [inline packages/dynamite/dynamite/lib/src/openapi_builder.dart:272]: This doesn't need to be a factory.

```dart
  UserStatusClient.fromClient(final DynamiteClient client)
      : super(
          client.baseURL,
          baseHeaders: client.baseHeaders,
          httpClient: client.httpClient,
          cookieJar: client.cookieJar,
          authentications: client.authentications,
        );
```

File: packages/nextcloud/lib/nextcloud.dart [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
 export 'package:crypton/crypton.dart' show RSAKeypair, RSAPrivateKey, RSAPublicKey;
 
-export 'src/nextcloud.openapi.dart' hide NextcloudClient;
+export 'src/api/core.openapi.dart' hide jsonSerializers, serializers;
+export 'src/api/news.openapi.dart' hide jsonSerializers, serializers;
+export 'src/api/notes.openapi.dart' hide jsonSerializers, serializers;
+export 'src/api/notifications.openapi.dart' hide jsonSerializers, serializers;
+export 'src/api/provisioning_api.openapi.dart' hide jsonSerializers, serializers;
+export 'src/api/uppush.openapi.dart' hide jsonSerializers, serializers;
+export 'src/api/user_status.openapi.dart' hide jsonSerializers, serializers;
```
Comment (Leptopoda) [inline packages/nextcloud/lib/nextcloud.dart:9]: we should annotate the hidden fields with @internal so we don't accidentally expose them in the future.

You probably need to add the `meta` package to nextcloud first.

**Review by Leptopoda — APPROVED** [review]

**Review by Leptopoda — APPROVED** [review]
Would it make sense to just make the serializer private and leave the default name but add a getter or other variable with the public name?

```dart

@SerializersFor([ ])
final Serializers _serializer = ...;

Serializer get userStatusSerializers => _serializer;
```

This way we wouldn't need to pass the name around all the time.

**Comment by provokateurin** [comment]
Oh that's a smart idea

**Review by Leptopoda — APPROVED** [review]

**Comment by provokateurin** [comment]
@Leptopoda Codeberg is down and now our CI is failing to build the docker image. I ran the whole pipeline locally, is it ok to merge?

**Comment by provokateurin** [comment]
Actually service seems to be restored just now, I'll give the CI another try

**Pull request merged**